src/sql/workbench/browser/modelComponents/button.component.ts

```diff
@@ -97,31 +109,43 @@ export default class ButtonComponent extends ComponentWithIconBase<azdata.Button
 	/// IComponent implementation
 
 	public setLayout(layout: any): void {
-		// TODO allow configuring the look and feel
 		this.layout();
 	}
 
 	public setProperties(properties: { [key: string]: any; }): void {
 		super.setProperties(properties);
-		this._button.enabled = this.enabled;
-		this._button.label = this.label;
-		if (this.properties.fileType) {
-			this.fileType = properties.fileType;
-		}
-		this._button.title = this.title;
+		if (this._informationalInputContainer) {
+			let button = this._button as InfoButton;
+			button.buttonMaxHeight = this.properties.height;
+			button.buttonMaxWidth = this.properties.width;
+			button.description = this.properties.description;
+			button.iconClass = createIconCssClass(this.properties.iconPath);
+			button.iconHeight = this.properties.iconHeight;
+			button.iconWidth = this.properties.iconWidth;
+			button.title = this.properties.title;
+		} else {
+			this._button.enabled = this.enabled;
+			this._button.label = this.label;
+
+			if (this.properties.fileType) {
+				this.fileType = properties.fileType;
+			}
+			this._button.title = this.title;
 
-		// Button's ariaLabel gets set to the label by default.
-		// We only want to override that if buttonComponent's ariaLabel is set explicitly.
-		if (this.ariaLabel) {
-			this._button.ariaLabel = this.ariaLabel;
-		}
+			// Button's ariaLabel gets set to the label by default.
+			// We only want to override that if buttonComponent's ariaLabel is set explicitly.
+			if (this.ariaLabel) {
+				this._button.ariaLabel = this.ariaLabel;
+			}
```

Comment:
please make sure for infobutton, the aria-label is set by its base classes and you have to set the aria-description attribute explicitly using the description property.

Reply:
I thought I left a comment about aria-label and aria-description, but couldn't find it.

Reply:
The info button does not need an aria-label because its text: title and description are present. These will instead be read by the screen reader. 

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute

Reply:
label and description DOM elements are inside the container, I believe when you turn on screen reader and do tab navigation, the description won't be read.

Reply:
The anchor will emit its contents to the screen reader. I just tested this using Voice Over on my Mac. Each button is identified first by its content and then the word "button" is spoken. This confirms my initial approach.